src/lib.rs

```diff
@@ -495,6 +495,43 @@ impl Client {
         nonce: Option<String>,
         signer: &impl Signer,
         url: &str,
+    ) -> Result<BytesResponse, Error> {
+        let mut retries = 3;
+
+        let mut nonce = nonce;
+        loop {
+            let mut response = self.inner_post(payload, nonce.clone(), signer, url).await?;
+            if response.parts.status == StatusCode::BAD_REQUEST {
```

Comment:
Nit: please invert the condition so we can do the early return here and reduce rightward drift in the rest of the loop body.